### PR TITLE
Reland: Migrate examples to the Android embedding v2

### PR DESCRIPTION
**Pull request by blasten**
Relands https://github.com/flutter/flutter/pull/43643 since https://github.com/flutter/engine/pull/13423 was merged and it's contained in https://github.com/flutter/flutter/pull/43860.

### CODE REVIEW

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt [hunk 1]
```diff
@@ -41,7 +44,7 @@ class MainActivity : FlutterActivity() {
 
         // Triggers the Android keyboard, which causes the resize of the Flutter view.
         // We need to wait for the app to complete.
-        MethodChannel(flutterView, "com.example.abstract_method_smoke_test")
+        MethodChannel(flutterEngine.getDartExecutor(), "com.example.abstract_method_smoke_test")
```
Comment (xster) [inline dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: @matthew-carroll these are all flutterEngine.getDartExecutor().getBinaryMessenger() now right?

**Review by xster — COMMENTED** [review]
LGTM, thanks!

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt (hunk 1 above)
Comment (blasten) [reply dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: yeah. I will follow up with that change.

**Comment by dnfield** [comment]
This appears to have broken the devicelab.  Going to try to revert to fix.

**Comment by xster** [comment]
Hard to tell from the logs. Wonder if 

```
**Error -32601 received from application: Method not found
{request: {method: ext.flutter.didSendFirstFrameRasterizedEvent, params: {isolateId: isolates/2092348583434603}}}**
```

is relevant @matthew-carroll

**Comment by dnfield** [comment]
Reverting it did make the devicelab green again.

**Comment by dnfield** [comment]
...excepting the dumb mac machine that forgot how to sign.

**Comment by xster** [comment]
> ...excepting the dumb mac machine that forgot how to sign.

what does this mean?

**Comment by dnfield** [comment]
My revert PR has two codesigning related failures in the devicelab from mac11.

**Comment by blasten** [comment]
@xster that is a red herring. 

In the [success case](https://flutter-dashboard.appspot.com/api/get-log?ownerKey=ahFmbHV0dGVyLWRhc2hib2FyZHJYCxIJQ2hlY2tsaXN0IjhmbHV0dGVyL2ZsdXR0ZXIvOGUwNzk5YTY1NzA1MWJmNTQwZDk4YjhhZWNmOWNmYTEzMjVmM2Y3MQwLEgRUYXNrGICAkI3UjokKDA),  the events `Flutter.Frame` and `Flutter.FirstFrame` are sent from the VM.  However, in the [failure case](https://flutter-dashboard.appspot.com/api/get-log?ownerKey=ahFmbHV0dGVyLWRhc2hib2FyZHJYCxIJQ2hlY2tsaXN0IjhmbHV0dGVyL2ZsdXR0ZXIvNzkzMjNmN2MxMzdhMjhmMzZhYzI3ZTU4OWY0YzgwZGM3Yzk2YzA3MgwLEgRUYXNrGICAkO3Fiv8IDA), they are not. As a result, the future never resolves.

**Comment by blasten** [comment]
@godofredoc is looking into reproducing this issue. For some reason, this issue doesn't repro locally and when @godofredoc tried to trigger the test manually in the device lab, it didn't repro either.

**Pull request merged**